### PR TITLE
AbDisc: Hit Selection Summary - scope to plate set path

### PR DESCRIPTION
**Pull request by labkey-nicka**
#### Rationale
Hit selection summary should respect lineage scope of plate set when aggregating hit selection results. This updates the backing query for "primary" plate set hit selection to scope to the "PlateSetPath" of the hit.

#### Related Pull Requests
- https://github.com/LabKey/labkey-ui-premium/pull/383
- https://github.com/LabKey/limsModules/pull/133
- https://github.com/LabKey/platform/pull/5421

#### Changes
- Calculate a path for each hit and persist to a new assay.Hit.PlateSetPath column
- Upgrade script for pre-existing hits

### CODE REVIEW

File: assay/src/org/labkey/assay/AssayUpgradeCode.java [hunk 1]
```diff
@@ -400,4 +405,67 @@ public static void deletePlateVocabDomains(ModuleContext ctx) throws Exception
...
+                    PlateSetLineage lineage = PlateManager.get().getPlateSetLineage(
+                        ContainerManager.getRoot(),
+                        User.getAdminServiceUser(),
+                        plateSetRowId,
+                        ContainerFilter.EVERYTHING
+                    );
+
+                    String lineagePath = lineage.getSeedPath();
+                    if (lineagePath == null)
+                        throw new ValidationException("");
```
Comment (labkey-klum) [inline assay/src/org/labkey/assay/AssayUpgradeCode.java]: Error message?

File: assay/src/org/labkey/assay/plate/model/PlateSetLineage.java [hunk 2]
```diff
@@ -95,4 +101,34 @@ public Map<Integer, PlateSet> getPlateSetAndDescendents(Integer plateSetId) {
 
         return allPlateSets;
     }
+
+    public @Nullable String getSeedPath() throws ValidationException
```
Comment (labkey-klum) [inline assay/src/org/labkey/assay/plate/model/PlateSetLineage.java]: I think a brief description on what the path represents, the direction of the IDs etc. would be helpful.

**Review by labkey-klum — APPROVED** [review]

File: assay/src/org/labkey/assay/AssayUpgradeCode.java (hunk 1 above)
Comment (labkey-nicka) [reply assay/src/org/labkey/assay/AssayUpgradeCode.java]: Revised so this is no longer necessary.

File: assay/src/org/labkey/assay/plate/model/PlateSetLineage.java (hunk 2 above)
Comment (labkey-nicka) [reply assay/src/org/labkey/assay/plate/model/PlateSetLineage.java]: Added.

Comment (labkey-klum) [reply assay/src/org/labkey/assay/plate/model/PlateSetLineage.java]: Thanks, looks good.

**Pull request merged**